code/modules/surgery/generic.dm

```diff
@@ -441,3 +441,52 @@
 	"<span class='warning'>Your hand slips, sawing through the bone in [target]'s [affected.display_name] with \the [tool]!</span>")
 	affected.createwound(CUT, 30)
 	affected.fracture()
+
+
+
+/////////BIOFOAM INJECTION///////
+/datum/surgery_step/generic/injectfoam/tool_quality(obj/item/tool)
+	. = ..()
+	if(!tool.is_sharp())
+		return 0
+
+/datum/surgery_step/generic/injectfoam
+	allowed_tools = list(
+	/obj/item/weapon/FixOVein/clot = 100,
+	)
+
+	priority = 0.1 //Tries to inject biofoam before other steps
+
+	min_duration = 10
+	max_duration = 20
+
+/datum/surgery_step/generic/injectfoam/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
```

Comment:
Need to check if the tool actually has foam to inject

Reply:
It can only do this step if it has sharpness and it olny has sharpness in injection mode.